### PR TITLE
Refresh docs index/landing page

### PR DESCRIPTION
**Pull request by rmoff**
Related to https://github.com/treeverse/lakeFS/issues/5617

Doesn't close out the above but takes a first pass at it. 

Focus is on getting the user into the hands-on quickly and moves the big scary diagram lower down the page

### CODE REVIEW

**Comment by rmoff** [comment]
@michalwosk adding you as reviewer as I think some of the content was previously added by you.

File: docs/index.md [hunk 1]
```diff
@@ -78,23 +88,23 @@ full list of commands.*
...
+<img src="/assets/img/promotion_workflow.png" alt="lakeFS promotion workflow" width="60%" height="60%" />
 
 1. First, create a new **branch** from `main` to instantly generate a complete "copy" of your production data.
 2. Apply changes or make updates on the isolated branch to understand their impact prior to exposure.
 3. And finally, perform a **merge** from the feature branch back to main to atomically promote the updates into production.
 
 Following this pattern, lakeFS facilitates a streamlined data deployment workflow that consistently produces data assets you can have total confidence in.
 
-## What else does lakeFS do?
+## How Can lakeFS Help Me?
```
Comment (michalwosk) [inline docs/index.md:103]: I am missing the dev\test use case. You are mentioning it clearly in the "git-for-data" section but I think it is missing an explicit reference here.

File: docs/index.md [hunk 2]
```diff
@@ -133,9 +143,9 @@ implement this logic yourself.
 
 Instead, make updates to the desired data assets on a branch and then utilize a lakeFS merge to atomically expose the data to downstream consumers.
 
-To learn more about atomic cross-collection updates, check out [this video](https://www.youtube.com/watch?v=9OsjUvk5UJU) which describes the concept in more detail.
+To learn more about atomic cross-collection updates, check out [this video](https://www.youtube.com/watch?v=9OsjUvk5UJU) which describes the concept in more detail, along with [this notebook](https://github.com/treeverse/lakeFS-samples/blob/main/01-multi-table-transaction-consistency-deltalake-lakefs/lakeFS-DeltaLake-multi-table-transaction-consistency.ipynb).
 
-#### Troubleshooting production problems
+### Troubleshooting production problems
```
Comment (michalwosk) [inline docs/index.md:148]: I am missing the ML reproduction, whether here or as a separate bullet.

File: docs/index.md [hunk 3]
```diff
@@ -50,21 +64,17 @@ df = spark.read.parquet("s3a://my-repo/main-branch/collections/foo/")
 
 You can use the same methods and syntax you are already using to read and write data when using a lakeFS repository. This simplifies the adoption of lakeFS - minimal changes are needed to get started, making further changes an incremental process.
```
Comment (michalwosk) [inline docs/index.md:66]: I think an internal link to the documentation of the integrations we have would be great here (I should have added it before already :-) )

File: docs/index.md [hunk 4]
```diff
@@ -1,36 +1,50 @@
...
+With lakeFS, you can use concepts on your data lake such as **branch** to create an isolated version of the data, **commit** to create a reproducible point in time, and **merge** in order to incorporate your changes in one atomic action.
 
-lakeFS is an open source project that supports managing data in  AWS S3, Azure Blob Storage, Google Cloud Storage (GCS) and any other object storage with an S3 interface. It integrates seamlessly with popular data frameworks such as Spark, Hive Metastore, dbt, Trino, Presto, and many others and even features an S3 compatibility layer.
+## How Do I Get Started? 
 
-The vision of lakeFS is bringing this functionality across all the data sources in your data pipelines, from analytics databases to key value stores - and to allow one system from which you can easily manage the underlying data in all data stores, with atomic Git-like actions.
+**[The hands-on quickstart](/quickstart) guides you through some of core features of lakeFS**. 
 
+These include [branching](/quickstart/branch.html), [merging](quickstart/commit-and-merge.html), and [rolling back changes](quickstart/rollback.html) to data.
```
Comment (michalwosk) [inline docs/index.md:26]: Maybe add a line here about the installation free cloud environment - which allows you to play around with a sample repo without installing anything.

**Review by michalwosk — CHANGES_REQUESTED** [review]
I think the change is awesome. It is very readable and provides clarity on lakeFS. I added some comments on the use cases section.

File: docs/index.md (hunk 2 above)
Comment (rmoff) [reply docs/index.md:148]: added

File: docs/index.md [hunk 5]
```diff
@@ -78,23 +88,23 @@ full list of commands.*
...
 Following this pattern, lakeFS facilitates a streamlined data deployment workflow that consistently produces data assets you can have total confidence in.
 
-## What else does lakeFS do?
+## How Can lakeFS Help Me?
 
-lakeFS helps you maintain a tidy data lake in several other ways, including:
+lakeFS helps you maintain a tidy data lake in several ways, including:
 
-#### Recovery from data errors
+### Recovery from data errors
```
[the remark this answers is not in the document]
Comment (rmoff) [reply docs/index.md:107]: moved

File: docs/index.md (hunk 1 above)
Comment (rmoff) [reply docs/index.md:103]: added

File: docs/index.md (hunk 3 above)
Comment (rmoff) [reply docs/index.md:72]: added

File: docs/index.md [hunk 6]
```diff
@@ -133,58 +164,25 @@ implement this logic yourself.
...
 
-Data engineers are often asked to validate the data. A user might report inconsistencies, question the accuracy, or simply report it to be incorrect. Since the data continuously changes, it is challenging to understand its state at the time of the error.
 
-The best way to investigate is by having a snapshot of the data as close as possible to the time of the error.
-Once you implement a regular commit cadence in lakeFS, each commit represents an accessible historical snapshot of the data. When needed, you may create a branch from a commit ID to debug an issue in isolation.
 
-To learn more on how to access a specific historical commit in a repository, see our seminal post on [data reproducibility](https://lakefs.io/solving-data-reproducibility/).
-
-#### Establishing data quality guarantees
+### Establishing data quality guarantees
```
Comment (michalwosk) [inline docs/index.md:171]: Can we change the title to: "Establishing data quality guarantees - CI / CD for data"
And as you did for the previous ones - add a link to here https://docs.lakefs.io/use_cases/cicd_for_data.html

**Review by michalwosk — APPROVED** [review]

**Pull request merged**